app/services/foreman/plugin.rb

```diff
@@ -351,6 +352,16 @@ def parameter_filters(klass)
       @parameter_filters.fetch(klass.is_a?(Class) ? klass.name : klass, [])
     end
 
+    def smart_proxy_for(klass, name, options)
+      @smart_proxies[klass.name] ||= {}
+      @smart_proxies[klass.name][name] = options
+      klass.register_smart_proxy(name, options)
```

Comment:
This probably won't work well with class autoloading, as if `klass` gets reloaded at any point the belongs_to association (etc) will be lost. You could either:

1. Add an `included` block to the concern so after it's loaded, it queries Foreman::Plugin to reload any smart proxies on the model.
1. Remove the `@smart_proxies` stuff (though it's my preferred method) and use `to_prepare` instead, which would be called on every development request (in the dev environment where reloading occurs).